cpp/src/lists/interleave_columns.cu

```diff
@@ -301,14 +353,21 @@ std::unique_ptr<column> interleave_columns(table_view const& input,
                  "All columns of the input table must be of lists column type.");
 
     auto const child_col = lists_column_view(col).child();
-    CUDF_EXPECTS(not cudf::is_nested(child_col.type()), "Nested types are not supported.");
     CUDF_EXPECTS(entry_type == child_col.type(),
                  "The types of entries in the input columns must be the same.");
   }
 
   if (input.num_rows() == 0) { return cudf::empty_like(input.column(0)); }
   if (input.num_columns() == 1) { return std::make_unique<column>(*(input.begin()), stream, mr); }
 
+  // For nested types, we rely on the `concatenate_and_gather` method, which costs more memory due
```

Comment:
This would be another good candidate for a hypothetical multi_gather function if we had it.  Each entry in the gather map would be a pair {column_index, row_index} selecting rows from different columns in an input table.

Reply:
Thanks Dave. The idea is great. I have filed an issue so we can track it here: https://github.com/rapidsai/cudf/issues/9175